tools/clang/unittests/HLSL/DxilModuleTest.cpp

```diff
@@ -598,4 +600,22 @@ TEST_F(DxilModuleTest, PayloadQualifier) {
                            DXIL::PayloadAccessShaderStage::Anyhit));
     }
   }
-}
\ No newline at end of file
+}
+
+TEST_F(DxilModuleTest, CanonicalSystemValueSemantic) {
+  // Verify that the semantic name is canonicalized.
+
+  // This is difficult to do from a FileCheck test because system value
+  // semantics get canonicallized during the metadata emit.  However, having a
+  // non-canonical semantic name at earlier stages can lead to inconsistency
+  // between the strings used earlier and the final canonicalized version.  This
+  // makes sure the string gets canonicalized when the signature elsement is
+  // created.
+
+  std::unique_ptr<hlsl::DxilSignatureElement> newElt = std::make_unique<hlsl::DxilSignatureElement>(hlsl::DXIL::SigPointKind::VSOut);
+  newElt->Initialize("sV_pOsItIoN", hlsl::CompType::getF32(),
```

Comment:
Do we have to support this?
It is really ugly.

Reply:
I'm not sure what you are referring to by "this".

Semantics are case-insensitive, but system values get canonicalized when written to metadata.  This just makes sure the system value is canonicalized when DxilSignatureElement is created, instead of waiting for the write to metadata.  That way any use of the semantic string between this point and the serialized form (specifically when writing the semantic string to RDAT) will get the same value as after serialization/deserialization.

Reply:
Sorry for not making things clearer.
I mean "sV_pOsItIoN" is ugly here.

Reply:
It's supposed to be case-insensitive.  It doesn't really matter what's here, as long as it's `"SV_Position"` with a different casing for any of the characters.  If it used `"SV_POSITION"` or `"sv_position"`, a different implementation might not properly do a case-insensitive match, instead checking for these more common possibilities only.